### PR TITLE
 Use Gradle Toolkit for testing Gradle plugin

### PR DESCRIPTION
**Pull request by dstepanov**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -147,6 +145,23 @@ project(':accurest-gradle-plugin') {
...
+	}
+
+    // Add the classpath file to the test runtime classpath
+	dependencies {
+		testRuntime files(createClasspathManifest)
+	}
+
 	// Hack for reusing this in the plugin test
 	test.dependsOn(':accurest-core:install')
 	test.dependsOn(':accurest-messaging-root:accurest-messaging-core:install')
```
Comment (marcingrzejszczak) [inline build.gradle]: WOW! Can you try removing those `test.dependsOn` and change them into:

```
compileOnly project(':accurest-messaging-root:accurest-messaging-integration')
```

and see if the build passes?

File: accurest-gradle-plugin/src/test/groovy/io/codearte/accurest/plugin/AccurestIntegrationSpec.groovy [hunk 2]
```diff
@@ -1,27 +1,57 @@
...
+import org.junit.rules.TemporaryFolder
+import spock.lang.Specification
 
 import java.nio.file.Files
 import java.nio.file.Path
 
 import static java.nio.charset.StandardCharsets.UTF_8
 
 /**
- * @author Olga Maciaszek-Sharma
```
Comment (szpak) [inline accurest-gradle-plugin/src/test/groovy/io/codearte/accurest/plugin/AccurestIntegrationSpec.groovy:12]: It is not a good habit to remove the original author ;). You could just append yourself.

File: build.gradle (hunk 1 above)
Comment (marcingrzejszczak) [inline build.gradle]: Ah and in the src/test/resources/...messaging.../build.gradle remove the line with importing of messaging:

https://github.com/Codearte/accurest/commit/ec1751fa935b261d7eba0ee7e8c40beb2af059f0

File: accurest-gradle-plugin/src/test/groovy/io/codearte/accurest/plugin/AccurestIntegrationSpec.groovy (hunk 2 above)
Comment (dstepanov) [inline accurest-gradle-plugin/src/test/groovy/io/codearte/accurest/plugin/AccurestIntegrationSpec.groovy:12]: Ops, I had a new file and then renamed it to AccurestIntegrationSpec, will fix it

**Comment by dstepanov** [comment]
I have deleted all extra dependencies and it passing because of `file("$outputDir/plugin-classpath.txt").text = sourceSets.test.runtimeClasspath.join("\n")`

**Comment by dstepanov** [comment]
Can somebody rerun the tests? Everything is passing locally :confused:

**Comment by szpak** [comment]
https://travis-ci.org/Codearte/accurest/jobs/124485688#L1732-L1745

```
    * What went wrong:
    Could not resolve all dependencies for configuration ':compileClasspath'.
    > Could not find any matches for io.codearte.accurest:accurest-messaging-integration:+ as no versions of io.codearte.accurest:accurest-messaging-integration are available.
      Searched in the following locations:
          https://repo1.maven.org/maven2/io/codearte/accurest/accurest-messaging-integration/maven-metadata.xml
          https://repo1.maven.org/maven2/io/codearte/accurest/accurest-messaging-integration/
          file:/home/travis/.m2/repository/io/codearte/accurest/accurest-messaging-integration/maven-metadata.xml
          file:/home/travis/.m2/repository/io/codearte/accurest/accurest-messaging-integration/
          http://repo.spring.io/snapshot/io/codearte/accurest/accurest-messaging-integration/maven-metadata.xml
          http://repo.spring.io/snapshot/io/codearte/accurest/accurest-messaging-integration/
          http://repo.spring.io/milestone/io/codearte/accurest/accurest-messaging-integration/maven-metadata.xml
          http://repo.spring.io/milestone/io/codearte/accurest/accurest-messaging-integration/
      Required by:
          :bootSimple:unspecified
```

You probably have it already installed in ~/.m2/repo

**Comment by marcingrzejszczak** [comment]
Did you remove `io.codearte.accurest:accurest-messaging-integration:+` build.gradle in the test project?

**Comment by dstepanov** [comment]
I have tried multiple combinations, the main problems is AccurestGradlePlugin which adds messaging dependency and the only way to fulfilled is to have it in repo.

**Comment by szpak** [comment]
That could be a problem - it is not a project level [dependency](https://github.com/Codearte/accurest/blob/master/accurest-gradle-plugin/src/main/groovy/io/codearte/accurest/plugin/AccurestGradlePlugin.groovy#L34) of the plugin, but it is added to end project directly.

Maybe [ivypot-gradle-plugin](https://github.com/ysb33r/ivypot-gradle-plugin) could be used to provide just built jar for tests?

**Comment by dstepanov** [comment]
There is one other way how to do it but needs investigating. Use add `flatDir` repository definition in Gradle and specify correct location to generated jars.

**Comment by marcingrzejszczak** [comment]
It's not that it's needed only for the part of the tests... I've added it here: https://github.com/Codearte/accurest/blob/master/accurest-gradle-plugin/src/main/groovy/io/codearte/accurest/plugin/AccurestGradlePlugin.groovy#L34 cause if you're using messaging I automatically add it. 

We can assume that if someone IS using messaging then he should manually add a required library (e.g. stream related) in his project. I think that it's a safe assumption.

@dstepanov - can you remove that line and rerun the tests?

**Comment by dstepanov** [comment]
All lines are needed :cry:

**Comment by dstepanov** [comment]
You meant messaging dependency? Not a dependency in build.gradle?

**Comment by marcingrzejszczak** [comment]
I'm already checking that fix and will push to this branch

**Comment by marcingrzejszczak** [comment]
Ok. I didn't manage to do that. I'd suggest merging what @dstepanov did, and then rewrite it with: https://github.com/ysb33r/ivypot-gradle-plugin . WDYT @szpak ?

**Comment by szpak** [comment]
+1

**Comment by szpak** [comment]
> We can assume that if someone IS using messaging then he should manually add a required library (e.g. stream related) in his project. I think that it's a safe assumption.

I don't know that new feature. Can AccuREST can be used without messaging?

If this is possible maybe it could be added by user in classpath scope or even a separate one accurest. See sample [usage](https://github.com/szpak/gradle-pitest-plugin#pit-plugins-support) from my plugin and [implementation](https://github.com/szpak/gradle-pitest-plugin/blob/master/src/main/groovy/info/solidsoft/gradle/pitest/PitestPlugin.groovy#L67).

**Comment by marcingrzejszczak** [comment]
@dstepanov  do you want to rewrite this to use the https://github.com/ysb33r/ivypot-gradle-plugin plugin Marcin has mentioned? :)

**Comment by dstepanov** [comment]
I would try to investigate flatDir approach ;)

**Comment by marcingrzejszczak** [comment]
Anything that will allow us to get ridd of the hacks :)

**Pull request merged**